### ISSUE TITLE
Animation is applied on sheet view and is propagated down to the content

### ISSUE THREAD

**Issue by NeverwinterMoon** [AndreaMiotto/PartialSheet#39]
There is this set on the sheet view:
`.animation(self.dragState.isDragging ? nil : .interpolatingSpring(stiffness: 300.0, damping: 30.0, initialVelocity: 10.0))` which applies animation to the content of the sheet. In my case, I have views which are hidden and shown depending on the state in the sheet content and now they are automatically animated. 

Per Apple's documentation on `animation(_:)`: `Use this modifier on leaf views rather than container views. The animation applies to all child views within this view; calling animation(_:) on a container view can lead to unbounded scope.`

Basically, setting this on the the container (sheet view) is an anti-pattern. I think, explicit property animation should be used instead, this way no matter what the content is provided to the PartialSheet, it will not be affected.

**Comment by AndreaMiotto** [comment]
Thank you for your feedback, can you provide a starting point with your code.
Anyway I will try to give it a look and modify it.

**Comment by NeverwinterMoon** [comment]
Hi!

My actual use case is slightly different and looks even worse, but this example hopefully gives an idea of what's wrong: https://github.com/NeverwinterMoon/PartialSheet-animation-issue

**Comment by AndreaMiotto** [comment]
Should be fixed, give it a try with v2.1.6

**Comment by NeverwinterMoon** [comment]
Well, this certainly does fix an issue when one doesn't want animations in the content view at all, but now it disables animations altogether for the content. Even if I use explicit property animations in the content view, they won't work, as the parent explicitly disables the animations for all the children views...

**Comment by AndreaMiotto** [comment]
ok you are right, but then I don't know how to fix this, I need to apply the animation on the parent:

```
.animation(self.dragState.isDragging ?
                        nil : .interpolatingSpring(stiffness: 300.0, damping: 30.0, initialVelocity: 10.0))
```

otherwise when you open and close the partial sheet the effect it's very bad

**Comment by NeverwinterMoon** [comment]
I tried doing a similar thing (to have nice effect for sheet appearing/disappearing) differently: by using .transition.

```
var body: some View {
    ZStack {
      if isOpen {
        background
          .transition(AnyTransition.opacity.animation(.easeInOut))

        GeometryReader { proxy in
          self.card(proxy: proxy)
        }
          .transition(.move(edge: .bottom))
          .onDisappear {
            self.opacity = 0.65
            self.dragStateClone = .inactive
          }
      }
    }
  }
```

I have separate transition for the slide up and for background fading in. Both are animated using property animation (on @Binding var isOpen: Bool), so they don't affect any children views.

**Comment by AndreaMiotto** [comment]
would you like to make a PR? at the moment I'm busy on other stuffs

**Comment by AndreaMiotto** [comment]
I really can't see how to apply your code to the partial sheet modifier

**Comment by NeverwinterMoon** [comment]
I could take a stab at a PR but it would also take me some time, as I am overly occupied with stuff at work.

Otherwise, the change might not be that trivial. The code I posted I've applied to a simplified version of PartialSheet and it is working really nicely with all the required animations on the sheet itself but without affecting the animations of the sheet content.

**Comment by styrken** [comment]
Hi @NeverwinterMoon, are you still up for the PR?

**Comment by 1337domz** [comment]
I have a somewhat working version which allows you to use implicit animations again (removes the .animation modifier):

https://github.com/1337domz/PartialSheet

At the moment:

- It will only work on iOS 14, as I am using the .onChange(of:) modifier. There is probably a way to do this differently to make it iOS 13 compatible.
- When the keyboard first gets shown it doesn't seem to adjust the height properly, I'm not sure why. I don't need the keyboard functionality so I didn't spend too much time trying to fix it.

Other than that, it seems to function as expected with the examples. I added a new example of an animation, that doesn't work with the currently released version. Maybe someone else can build upon this?

**Comment by eliottrobson** [comment]
I noticed that the animations were clunky when showing and hiding the sheet so I also started to have a look at this. I have a working version with explicit animations in that pull request. I hope this helps us get this issue closed.

**Comment by styrken** [comment]
Awesome guys! I will have a look at it
